include/picongpu/plugins/transitionRadiation/TransitionRadiation.hpp

```diff
@@ -449,6 +528,204 @@ namespace picongpu
                     }
                 }
 
+                void writeOpenPMDFile(uint32_t currentStep)
+                {
+                    auto buffer = std::make_unique<HostBuffer<float_X, DIM3>>(DataSpace<DIM3>(
+                        parameters::nTheta,
+                        parameters::nPhi,
+                        transitionRadiation::frequencies::nOmega));
+                    auto dataBox = buffer->getDataBox();
+
+                    for(unsigned int index_direction = 0; index_direction < transitionRadiation::parameters::nObserver;
+                        ++index_direction)
+                    {
+                        // theta
+                        const int i = index_direction / parameters::nPhi;
+                        // phi
+                        const int j = index_direction % parameters::nPhi;
+
+                        for(unsigned int k = 0; k < transitionRadiation::frequencies::nOmega; ++k)
+                        {
+                            dataBox({static_cast<int>(i), static_cast<int>(j), static_cast<int>(k)})
+                                = static_cast<float_X>(
+                                    theTransRad[index_direction * transitionRadiation::frequencies::nOmega + k]);
+                        }
+                    }
+
+                    std::stringstream filename;
+                    filename << filenamePrefix << "_%T." << fileExtension;
+
+                    ::openPMD::Series series(filename.str(), ::openPMD::Access::CREATE);
+
+                    ::openPMD::Extent extent
+                        = {static_cast<unsigned long int>(transitionRadiation::frequencies::nOmega),
+                           static_cast<unsigned long int>(parameters::nPhi),
+                           static_cast<unsigned long int>(parameters::nTheta)};
+                    ::openPMD::Offset offset = {0, 0, 0};
+                    ::openPMD::Datatype datatype = ::openPMD::determineDatatype<float_X>();
+                    ::openPMD::Dataset dataset{datatype, extent};
+
+                    auto mesh = series.iterations[currentStep].meshes["transitionradiation"];
+
+                    mesh.setAxisLabels(std::vector<std::string>{"omega index", "phi index", "theta index"});
+                    mesh.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    mesh.setGridUnitSI(1);
+                    mesh.setGridSpacing(std::vector<double>{1, 1, 1});
```

Comment:
I take it that this is the use case you mentioned a while ago where the mesh is not purely spatial? Otherwise, this specifies that one cell is a 1m cube.

Reply:
Yes, this grid has different units depending on the axis and I tried to use the radiation-plugin approach by having a separate grid for every axis, which determines the unit.

Reply:
Discussed today: We will leave this for now as I plan to implement https://github.com/openPMD/openPMD-standard/pull/193 soon which will help specifying this more precisely.